HPS/Heartland/view/adminhtml/templates/form/cc.phtml

```diff
@@ -1,151 +1,285 @@
+                            }
+                        },
+                        deleteCard: function (id,e) {//{"token_value":"3","cc_last4":"1111","cc_type":"visa",
+                            // "cc_exp_month":"2","cc_exp_year":"2025"}
+                            $("#iframes").fadeOut();                            
+                            if(confirm('Are you sure you want to delete the stored card?')){
+                                $.ajax({
+                                    url: "<?php echo $block->getBaseUrl() ?>/heartland/creditcard/delete?t="+id
+                                    , success: function (data) {
+                                            e.remove();
+                                            alert('Your saved card has been removed from our records ');
+                                    }
+                                });
+                            }
+
+                        },
+
+                        hpsNewCard: function () {
+                            $('#securesubmit_token').removeAttr('value');
+                            ;
+                            $("#SelectNewCardHPS").prop('checked', true);
+                            <?php /* @noEscape */ echo $code; ?>.
+                            hpsBusy();
+                            <?php /* @noEscape */ echo $code; ?>.
+                            hpsShowCcForm('<?php /* @noEscape */ echo $pubApiKey?>');
+                            return true;
+                        },
+
+                        hpsBusy: function () {
+                            $("#checkout-loader-iframeEdition").fadeIn();
+                        },
+
+                        hpsNotBusy: function () {
+                            $("#checkout-loader-iframeEdition").fadeOut();
+                            _HPS_EnablePlaceOrder()
+                        },
+
+                        hpsShowCcForm: function (publicKey) {
+                            if (publicKey) {
+
+                                $("#iframes").fadeIn();
+                                HPS_SecureSubmit($,document, Heartland, publicKey);
+
+
+                            }
+
+                        }
+                        ,
+                        hideNewCardForm: function () {
+                            $("#iframes").fadeOut();
+                        },
+
+                        getCode: function () {
+                            try {
+                                $("#iframes").fadeOut();
+                            } catch (e) {
+                            }
+                            return '<?php /* @noEscape */ echo $code; ?>';
+
+                        },
+
+                        isActive: function () {
+                            return true;
+                        },
+
+
+
+
+                        }
+                });
+
+
+            }
+        );
+
+
+    </script>
+    <table id="SavedCardsTable">
```

Comment:
even though there is no saved card, we are having a tr tag for "New Card / Update Saved Card". So, this table tag is needed here